### PR TITLE
Hide reference price if referenceUnit = 0

### PR DESCRIPTION
**Pull request by pascalheidmann-bedarf**
### 1. Why is this change necessary?
Without this PR price calculations per unit result in 0,00/unit prices.

### 2. What does this change do, exactly?
Add check to check if sArticle.referenceunit is defined if not hide those calculations.

### 3. Describe each step to reproduce the issue or behaviour.
1. Create/edit an article to have purchaseunit set to 1 (or anything other) but leave referenceunit empty.
2. Load up articles detail page on frontend
3. Without this PR it will state something like _1 piece (0.00€ / 1 piece)_ below the price. With this PR this infobox won't show up.

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by pascalheidmann-bedarf** [comment]
As an alternative you might set referenceunit=1 if referenceunit is not a valid number.

**Comment by Mardl** [comment]
Missing 
themes/Frontend/Bare/frontend/detail/block_price.tpl:68
`{if $hasReferencePrice && $sArticle.referenceunit }`

File: themes/Frontend/Bare/frontend/detail/block_price.tpl [hunk 1]
```diff
@@ -65,7 +65,7 @@
                                                         {$blockPrice.price|currency} {s name="Star" namespace="frontend/listing/box_article"}{/s}
                                                     </td>
                                                 {/block}
-                                                {if $hasReferencePrice}
+                                                {if $hasReferencePrice && $sArticle.referenceunit }
```
Comment (mitelg) [inline themes/Frontend/Bare/frontend/detail/block_price.tpl]: would it make sense to add this to [Line 7](https://github.com/shopware/shopware/pull/1573/files#diff-f7419b3e9dcd913aafc5943b9c96bbb0R7)?

Comment (Mardl) [reply themes/Frontend/Bare/frontend/detail/block_price.tpl]: oh, yes, sorry. Need in line 7, because it is also relevant for line 27

**Comment by soebbing** [comment]
Hi @pascalheidman-bedarf, thank you for your contribution! 💙

**Pull request closed without merging**